### ISSUE TITLE
HMSAnalyticsManager doesn't track custom events

### ISSUE THREAD

**Issue by gtoto007** [EvilMindDevs/hms-unity-plugin#175]
Hi, 

I'm a developer of Marshmallow Games SRL. We have released our app Smart Tales on store appgallery "https://appgallery.huawei.com/#/app/C104663029" .

I am using the HMS unity 2.0.10 and I ran test scene  **AnalyticsDemo** from the P40 device but **when I launch a custom event  it isn't tracked in real time**. Infact, I can only see  the _"$InstallApp"_ and _"$UninstallApp"_ native events tracked in console.

![Schermata 2021-09-01 alle 16 24 36](https://user-images.githubusercontent.com/6715157/131688888-45a2a537-6769-4dc5-84e0-fb47bdd5d209.png)

I **tracked the logs with adb logcat** when I launch the custom event "t" via AnalyticsDemo scene.

```
 Unity  I  [HMS] : Analytics Kits Event Id:t Key:t Value:t
                         I  HMSAnalyticsManager:SendEventWithBundle(String, Stri
                            ng, String)
                         I  UnityEngine.Events.UnityAction:Invoke()
                         I  UnityEngine.Events.UnityEvent:Invoke()
                         I  UnityEngine.EventSystems.EventFunction`1:Invoke(T1, 
                            BaseEventData)
                         I  UnityEngine.EventSystems.ExecuteEvents:Execute(GameO
                            bject, BaseEventData, EventFunction`1)
                         I  UnityEngine.EventSystems.StandaloneInputModule:Proce
                            ssTouchPress(PointerEventData, Boolean, Boolean)
                         I  UnityEngine.EventSystems.StandaloneInputModule:Proce
                            ssTouchEvents()
                         I  UnityEngine.EventSystems.StandaloneInputModule:Proce
                            ss()
```

I have already **activated Huawei Analytics** in the project settings on appgallery, **updated agconnect-services.json** in Assets/streamingAssets and **activated Analytics in the Unity Project Hms** settings.

![Schermata 2021-09-01 alle 16 13 28](https://user-images.githubusercontent.com/6715157/131687042-f6e516b5-d4da-41b1-a47c-395e0ada30c4.png)

My agconnect-services.json file in streamingAssets
[agconnect-services.json.txt](https://github.com/EvilMindDevs/hms-unity-plugin/files/7091777/agconnect-services.json.txt)
HMS SETTINGS 
<img width="817" alt="Schermata 2021-09-01 alle 16 18 35" src="https://user-images.githubusercontent.com/6715157/131687974-8851afb7-0714-434e-8251-da2b9a8d09d8.png">

I also **created a custom event** on the app gallery store project.
![Schermata 2021-09-01 alle 16 17 30](https://user-images.githubusercontent.com/6715157/131687728-1b2616f2-ae1c-478e-b7d2-f0042dd4e120.png)

**Comment by sametguzeldev** [comment]
Hello there.
I would suggest you use the App Debugging feature for testing analytics.
I have written an article covering HMS Analytics. You can check it out [here](https://medium.com/huawei-developers/boost-your-game-with-huawei-analytics-in-unity-8e4b02702edf).

Also, sometimes it caches all analytics data and sends it to Huawei servers when the application gets closed. This behavior is to reduce the traffic between the user and the server.

**Comment by gtoto007** [comment]
Thank you for your tutorial!
I have activated App Debugging mode with 
`adb shell setprop debug.huawei.hms.analytics.app com.marshmallowgames.smarttales.huawei`
and I can see my customs in App Debug 
![Schermata 2021-09-01 alle 19 17 04](https://user-images.githubusercontent.com/6715157/131716295-a7b9fb99-c991-4998-9f85-4e8129f37506.png)

But When I disabled the App Debugging mode, I don't see my custom events in **Real Time overviews**
![Schermata 2021-09-01 alle 19 28 19](https://user-images.githubusercontent.com/6715157/131716818-1d1ee899-7d0a-40eb-bdb9-579f8614f1c0.png)

**Comment by gtoto007** [comment]
I am noticing that a custom event is tracked always with an extra parameter which is called $TaskId and has an empty value. 

![Schermata 2021-09-01 alle 19 53 59](https://user-images.githubusercontent.com/6715157/131720380-fd74c641-9ce6-4ec1-94bf-6a42dc69b40f.png)

I wondered if the fact that it is not being tracked is because I registered the custom event with one parameter and instead he expects to receive the event with two parameters. If so, I do not have the possibility to add the second parameter $TaskId to my custom event. Because when I add  it, the system say me that the character $ is invalid in event name.

**Comment by sametguzeldev** [comment]
I think that parameter is getting added by HMS Analytics for reports. You don't need to worry about it.

**Comment by gtoto007** [comment]
Ok,  but I don't understand the problem. You think that  huawei services  have some issue?

**Comment by sametguzeldev** [comment]
I will run some tests and come back to you with results today.

**Comment by gtoto007** [comment]
The app with Kit Analytics has just been approved and now the customs events are tracking correctly in real time mode! 
so I assume that until the app is approved, the events are not tracked? Or was there some disruption with huawei services? 
![Schermata 2021-09-02 alle 18 13 35](https://user-images.githubusercontent.com/6715157/131880391-5fea9121-4859-4fbf-af8a-cec6e2c35ff5.png)

**Comment by sametguzeldev** [comment]
I just tried it and it did work for me.
I haven't changed any code. Sent "x" and "m" parameters similiar as you did. I did register "x" on events but I did not register "m". Both of them still were on my analytics.
This is without App Debugging btw. 
![image](https://user-images.githubusercontent.com/6827857/131887290-ed691f62-c57b-4566-8f31-e53d5b311404.png)
![image](https://user-images.githubusercontent.com/6827857/131887052-ea4b2398-03ff-4b85-af28-edf447bc3120.png)
